android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModel.kt

```diff
@@ -588,19 +589,17 @@ constructor(
       .all { it is Valid || it is NotValidated }
   }
 
-  suspend fun executeCql(subject: Resource, bundle: Bundle, questionnaire: Questionnaire) {
+  suspend fun executeCql(
+    subject: Resource,
+    bundle: Bundle,
+    questionnaire: Questionnaire,
+    questionnaireConfig: QuestionnaireConfig? = null,
+  ) {
+    questionnaireConfig?.cqlInputResources?.forEach { resourceId ->
+      val basicResource = defaultRepository.loadResource(resourceId) as Basic?
+      bundle.addEntry(Bundle.BundleEntryComponent().setResource(basicResource))
+    }
```

Comment:
can you please add a test that loads cqlInputResources from a questionnaireConfig

Reply:
I have added a simple test `testLoadCqlInputResourcesFromQuestionnaireConfig` which checks if QuestionnaireConfig is properly loading the cqlInputResources list, after initialisation
Let me know if there is anything else that needs to be added.

Reply:
cool so in that test can you then call executeCql?

You can mock the libraryEvaluator.runCqlLibrary call, intercept the bundle and verify that the cqlInputResources have been added to the bundle